whole-tale/dashboard
# Publish modal styling changes

### Problem
The Publish Modal is inconsistent with the styling elsewhere in the application.

Fixes #446

### Approach
Refactor look and feel of the Publish Modal slightly to better match other parts of the WholeTale Dashboard. This also includes some refactoring / better error-handling for the `notification-stream`, as we plan to soon enhance the publishing task to provide asynchronous updates via Girder's Notification API.

### How to Test
1. Checkout and run this branch locally, rebuild the dashboard
2. Open a new incognito window and login to the WholeTale Dashboard
3. Compose and Launch a Tale, if you haven't already - ensure that this is a Tale that you own
    * NOTE: Once #448 is merged, you should no longer be given an option to Launch tales that you do not own
4. Navigate to Run and select a Tale
5. Click the ellipses on the `wt-paddleboard` to open a dropdown, then click "Publish Tale..."
    * You should see a modal appear, prompting / redirecting you to go to the DataONE login page
    * Once logged in, you should be redirected back to the WholeTale Dashboard (with `?auth=true` appended to the URL)
6. Navigate once again to Run, select a Tale, expand the dropdown, and click "Publish Tale..."
    * This time, the Publish Modal should appear in the view
    * The "Target Repository" dropdown should automatically expand for you, saving you a mouse click
7. Select "DataONE Development" from the dropdown and click "Publish"
    * You should see the "Publish" button change to "Publishing", and its checkmark should change to a loading symbol - the button should be disabled at this time
    * You should see a progress bar appear in a blue message at the bottom of the modal
8. Wait for the publishing operation to complete successfully
    * You should see the "Publishing" button change to "Published", and its checkmark should reappear - the button should still be disabled, preventing a 2nd publish without reopening the modal
    * You should see the blue progress message has been replaced with a green success message

NOTE: In the case of error, the green success message is replaced with a red error message. This error message should display any error that comes back from the publishing API or Job.
